invenio_oaiserver/query.py

```diff
@@ -45,6 +47,24 @@ def match(self, record):
         return self.query.accept(MatchUnit(record))
 
 
+def get_affected_records(spec=None, search_pattern=None):
+    """Get list of affected records."""
+    sets = '_oai.sets:"{0}"'.format(spec) if spec else None
+    search_pattern = search_pattern if search_pattern else None
```

Comment:
If `search_pattern` is `None` then the OAI set is manually managed, hence it should not be updated.